pkg/azuredisk/azuredisk.go

```diff
@@ -398,3 +395,18 @@ func getSourceVolumeId(snapshot *compute.Snapshot) string {
 	}
 	return ""
 }
+
+func getValidCreationData(subscriptionID, resourceGroup, sourceResourceID string) compute.CreationData {
```

Comment:
may return error if it could not get a valid snapshot URL.

Reply:
done.